core/locale/infrastructure/translation_service.go

```diff
@@ -4,22 +4,24 @@ import (
 	"bytes"
 	"errors"
 	"fmt"
+	"os"
+	"sync"
 	"text/template"
+	"time"
 
 	"flamingo.me/flamingo/v3/core/locale/domain"
-
 	"flamingo.me/flamingo/v3/framework/config"
 	"flamingo.me/flamingo/v3/framework/flamingo"
 	"github.com/nicksnyder/go-i18n/i18n/bundle"
 )
 
 // TranslationService is the default TranslationService implementation
 type TranslationService struct {
-	translationFile  string
-	translationFiles config.Slice
+	sync.RWMutex
```

Comment:
I would use a named private field here, otherwise we expose all the Lock/Unlock methods and I don't think we want outside users to lock the TranslationService.

Reply:
You're right, I changed it.